src/Trainer.py

```diff
@@ -64,58 +69,15 @@ def get_lr(step):
                       f"Scale: {model.logit_scale.exp().item():.1f}"
                       )
             global_step +=1
-            torch.save({
-            'epoch': epoch,
-            'model_state_dict': model.state_dict(),
-            'optimizer_state_dict': optimizer.state_dict(),
-            'loss': loss.item(),
-            #'scaler': scaler.state_dict() if enable_amp else None,
-        }, f'clip_checkpoint_epoch_{epoch}.pt')
+            if epoch%5 ==0:
+                torch.save({
+                'epoch': epoch,
+                'model_state_dict': model.state_dict(),
+                'optimizer_state_dict': optimizer.state_dict(),
+                'loss': loss.item(),
+                #'scaler': scaler.state_dict() if enable_amp else None,
+            }, f'{dest_folder}/clip_checkpoint_epoch_{epoch}.pt')
         
             print(f"Epoch {epoch+1} completed. Average Loss: {total_loss / num_batches:.4f}")
     
-    return model
-
-
-            
-
-class Image_Text_Dataset():
```

Comment:
removed the Image_Text_Dataset, but incorporated its functionality into `io.py` where the data loader works directly on a single mosaic fits file.